db/document.go

```diff
@@ -38,6 +39,171 @@ const (
 	DocUnmarshalNone                                     // No unmarshalling (skips import/upgrade check)
 )
 
+// Struct populated by an incoming document
+type IncomingDocument struct {
```

Comment:
What's the contract here for _body and RawBody? Do either of them contain stamped special properties?